### PR TITLE
MM-T2503 Boolean value check

### PR DESCRIPTION
**Pull request by MikeworX**
**_Summary_**

This PR checks that true/false values returned by the demo-plugin's /dialog form are indeed boolean. 

**_Steps_**

- the form is completed the first time with all values checked as true
- once the form is submitted, the true values are asserted to be true (not "true")
- the above is then repeated for false values

Screen Shot 2020-11-03 at 9 25 14 PM

**_Links_**

[vercel](https://automation-test-cases.vercel.app/test/MM-T2503)
[TM4J](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T2503)

### CODE REVIEW

**Review by srkgupta — CHANGES_REQUESTED** [review]
Thanks @MikeworX. Instead of duplicating the steps in the testcase, can you write a function which accepts true/false as a parameter and then performs the steps accordingly.

File: e2e/cypress/integration/interactive_dialog/boolean_value_check_spec.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        cy.apiInitSetup().then(({team, user, channel}) => {
+            testChannel = channel;
+
+            // # Set up Demo plugin
+            cy.apiInstallPluginFromUrl('https://github.com/mattermost/mattermost-plugin-demo/releases/download/v0.8.0/com.mattermost.demo-plugin-0.8.0.tar.gz', true);
+            cy.apiEnablePluginById('com.mattermost.demo-plugin');
+
+            // # Login as regular user
+            cy.apiLogin(user);
+            cy.visit(`/${team.name}/channels/${testChannel.name}`);
```
Comment (srkgupta) [inline e2e/cypress/integration/interactive_dialog/boolean_value_check_spec.js:26]: Add an additional step to wait till postbox is visible. 
`cy.get('#post_textbox', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible');`

**Comment by MikeworX** [comment]
Hi @srkgupta - thank-you for reviewing. Are you referring to the 2 assertion steps or the form setup?

**Comment by srkgupta** [comment]
> Hi @srkgupta - thank-you for reviewing. Are you referring to the 2 assertion steps or the form setup?

I mean from the steps from line 37 to 75 are almost the same except for the condition true or false. Hence you can just have one function and accept one parameter (true or false) and accordingly set the form values and then validate the tests, instead of duplicating these steps

**Comment by saturninoabril** [comment]
I agree with Rohitesh. Ping me whenever ready for review.

**Pull request closed without merging**